### DISCUSSION TITLE
Unmyelinated axons

### DISCUSSION THREAD

**Discussion by Velautham** [neuropoly/axondeepseg#688] [Help needed]
How would we also include unmyelinated/demyelinated axons in our segmentation mask for research purposes? Any pre-existing models? Please advise.
Thanks

**Comment by mathieuboudreau** [comment]
Hi @Velautham - do you have any sample images (& resolution) that you could share, so that we may know more about your dataset use-ase?

**Reply by Velautham** [reply]
Yes, we do Dr. Boudreau. Will it be possible to continue correspondence over email?

**Reply by mathieuboudreau** [reply]
Yes - you can email axondeepseg@googlegroups.com

**Reply by mathieuboudreau** [reply]
Hi @Velautham - we've received your images. Do you by any chance have an annotated/manually segmented image identifying the problematic / unmyelinated axons? I'm not a biologist and at first glance the axons mostly look well myelinated to me =)

**Reply by mathieuboudreau** [reply]
Also maybe, the output from AxonDeepSeg if you tried it already, and highlight which axons did not segment to your satisfaction

**Reply by Velautham** [reply]
Hi, sorry for the delayed response. I have sent an email with the information.

**Reply by ktyssowski** [reply]
Hi! I'm also interested in segmenting unmyelinated axons and I'm wondering if anything ever became of this. Thanks!

**Reply by hermancollin** [reply]
Hi @ktyssowski . Thank you for your interest.

Indeed, this project is ongoing. Annotations were created and a first round of training was completed. We are currently collecting data with another group for a second round of training. 

You can actually use the model at its current state - see https://github.com/axondeepseg/nn-axondeepseg. Currently, the model works relatively well for TEM images but still slightly undersamples unmyelinated axons. I'd be happy to hear your feedback if you want to try it.

**Reply by ktyssowski** [reply]
great! thanks! i'll try that -- just to make sure i understand, I should download nn-axondeepseg to use through the command line, and that's different from the deeplearning axondeepseg that i'm using through napari/the command line? it's not just a new model i can add to what i'm already using?

**Reply by hermancollin** [reply]
Ah yes, I should have specified this. What's your level of comfort with Python in general?

The `nn-axondeepseg` repository I just sent you is only a temporary place to put some new models and let collaborators use them before we actually add them into the axondeepseg software you are used to. The plan is most definitely to support it in the main software soon, but for the moment you would have to use this command line tool.

**Reply by ktyssowski** [reply]
i'm fairly comfortable with python, and i can run it through the command line -- i just like the napari interface for convenience and being able to edit
